docs/content/guides/dagster/branch_depl_with_snowflake.mdx

```diff
@@ -0,0 +1,325 @@
+
+- A running agent to serve your branch deployments
+- A Dagster project set up for branch deployments containing the following:
+  - `branch_deployments.yml` GitHub workflow file
+  - Dagster repository (we’ve named ours `repo`)
+  - Dockerfile that installs your Dagster project
+
+If you’re new to Dagster, here’s an overview of the main concepts we’ll be using:
+
+- [Asset](https://docs.dagster.io/concepts/assets/software-defined-assets) - an asset is a software object that models a data asset. The prototypical example is a table in a database or a file in cloud storage.
+- [Resource](https://docs.dagster.io/concepts/resources) - a resource is an object that models a connection to a (typically) external service. Resources can be shared between assets and different implementations of resources can be swapped in depending on environment. For example, a resource may provide methods to send messages in Slack.
+- [IO manager](https://dagster-git-jamie-local-to-prod-example-elementl.vercel.app/concepts/io-management/io-managers) - an IO manager is a special kind of resource that handles storing and loading assets. For example, if we wanted to store assets in S3, we would use Dagster’s built in S3 IO manager.
+- [Op](https://docs.dagster.io/concepts/ops-jobs-graphs/ops) - an op is a unit of computation that accepts inputs and passes outputs to other ops.
+- [Graph](https://docs.dagster.io/concepts/ops-jobs-graphs/graphs) - a graph is an interconnected set of ops.
+- [Job](https://docs.dagster.io/concepts/ops-jobs-graphs/jobs#jobs) - a job is the main unit of execution in Dagster. Internally, it is a graph bound to a set of resources and execution configuration.
+
+### Overview
+
+We have a `PRODUCTION` Snowflake database with a schema named `HACKER_NEWS`. In our production cloud environment, we’d like to write tables to Snowflake containing subsets of Hacker News data. These tables will be:
+
+- `items` (the entire dataset)
+- `comments` (data about comments)
+- `stories` (data about stories)
+
+In order to set up a branch deployment workflow to construct and test these tables, we will:
+
+1. Define these tables as [assets](https://docs.dagster.io/concepts/assets/software-defined-assets).
+2. Configure our assets to write to Snowflake using a different connection (credentials and database name) for two environments: production and branch deployment.
+3. Clone our production database upon each branch deployment launch, naming each clone `PRODUCTION_CLONE_<ID>`, where `<ID>` is the pull request ID of the branch. We’ll create a branch deployment to test our Hacker News assets.
```

Comment:
> We'll create a branch deployment to test our Hacker News assets

I think this could be step 4 (and then the next step would be step 5)

Reply:
circling back to this - i see now that the numbers of these steps correspond to the sections in the guide. I don't think creating a branch deployment is worthy of it's own section (it fits nicely with section 3) but maybe the sentence here could be something like "Then we'll create a branch deployment and test our Hacker News assets against our newly cloned database"